pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -821,25 +823,64 @@ func (r *Builder) TemplateLabels(vmRef ref.Ref) (labels map[string]string, err e
 		os = UnknownOS
 	}
 
-	if os != UnknownOS && version != "" {
-		os = fmt.Sprintf("%s%s", os, version)
-		if distro == CentOS && len(version) >= 1 && (version[:1] == "8" || version[:1] == "9") {
-			os = fmt.Sprintf("%s-stream%s", distro, version)
-		} else if os == Windows {
-			os = DefaultWindows
-			if strings.Contains(version, "2k12") || strings.Contains(version, "2012") {
-				os = fmt.Sprintf("%s2k12", os)
-			} else if strings.Contains(version, "2k16") || strings.Contains(version, "2016") {
-				os = fmt.Sprintf("%s2k16", os)
-			} else if strings.Contains(version, "2k19") || strings.Contains(version, "2019") {
-				os = fmt.Sprintf("%s2k19", os)
-			} else if strings.Contains(version, "2k22") || strings.Contains(version, "2022") {
-				os = fmt.Sprintf("%s2k22", os)
-			} else if len(version) >= 2 && version[:2] == "11" {
-				os = fmt.Sprintf("%s%s", os, version)
+	if template {
+		if os != UnknownOS && version != "" {
+			os = fmt.Sprintf("%s.%s", os, version)
+			if distro == CentOS && len(version) >= 1 && (version[:1] == "8" || version[:1] == "9") {
+				os = fmt.Sprintf("%s-stream%s", distro, version)
+			} else if os == Windows {
+				os = DefaultWindows
+				if strings.Contains(version, "2k12") || strings.Contains(version, "2012") {
+					os = fmt.Sprintf("%s2k12", os)
+				} else if strings.Contains(version, "2k16") || strings.Contains(version, "2016") {
+					os = fmt.Sprintf("%s2k16", os)
+				} else if strings.Contains(version, "2k19") || strings.Contains(version, "2019") {
+					os = fmt.Sprintf("%s2k19", os)
+				} else if strings.Contains(version, "2k22") || strings.Contains(version, "2022") {
+					os = fmt.Sprintf("%s2k22", os)
+				} else if len(version) >= 2 && version[:2] == "11" {
+					os = fmt.Sprintf("%s%s", os, version)
+				}
+			}
+		}
+	} else {
+		if os != UnknownOS && version != "" {
```

Comment:
what's the difference between those blocks (templates vs. instance types)?
if there's a difference, I'd extract those to separate functions rather than passing a boolean argument for indicating whether we look for a template or an instance type

Reply:
+1

Looks like this can be deduplicated.